x-pack/filebeat/input/s3/config.go

```diff
@@ -42,16 +44,22 @@ func (c *config) Validate() error {
 		return fmt.Errorf("visibility timeout %v is not within the "+
 			"required range 0s to 12h", c.VisibilityTimeout)
 	}
+
 	if c.APITimeout < 0 || c.APITimeout > c.VisibilityTimeout/2 {
 		return fmt.Errorf("api timeout %v needs to be larger than"+
 			" 0s and smaller than half of the visibility timeout", c.APITimeout)
 	}
+
 	for i := range c.FileSelectors {
 		r, err := regexp.Compile(c.FileSelectors[i].RegexString)
 		if err != nil {
 			return err
 		}
 		c.FileSelectors[i].Regex = r
 	}
+
+	if c.MaxNumberOfMessages > 10 || c.MaxNumberOfMessages < 1 {
+		return fmt.Errorf(" max_number_of_messages %v needs to be between 1 and 10", c.MaxNumberOfMessages)
```

Comment:
Not familiar with this setting effects but I guess these values are sane? Wondering it `10` is just too small and might lead to overloading the API with too many requests.

Reply:
@ChrsMark MaxNumberOfMessages documentation `Valid values: 1 to 10`: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html 

Yeah unfortunately this is set by AWS. But in AWS, the default is 1 but I'm using 5 as the default. Originally before this change, we are hardcoding this value with 10. But some users are seeing some perf issue. That's why I'm changing the default to 5 and make it also configurable.